libs/core/src/lib/select/option/option.component.ts

```diff
@@ -1,22 +1,61 @@
 import {
+    AfterViewChecked,
     ChangeDetectionStrategy,
     ChangeDetectorRef,
     Component,
     ElementRef,
-    HostBinding,
-    HostListener,
+    EventEmitter,
+    HostBinding, InjectionToken,
     Input,
     OnDestroy,
-    OnInit,
+    Output,
     ViewEncapsulation
 } from '@angular/core';
-import { filter, map } from 'rxjs/operators';
-import { Subscription } from 'rxjs';
-import { KeyUtil } from '../../utils/functions/key-util';
-import { SelectProxy } from '../select-proxy.service';
+import { Subject } from 'rxjs';
+import { ActiveDescendantKeyManager, FocusableOption, FocusOrigin } from '@angular/cdk/a11y';
+import { ENTER, hasModifierKey, SPACE } from '@angular/cdk/keycodes';
+import { FdSelectChange } from '../select.component';
+import { DialogConfig, MobileModeConfig } from '../../..';
 
 let optionUniqueId: number = 0;
 
+
+/**
+ * Event object emitted by OptionComponent when selected or deselected. *
+ */
+export class FdOptionSelectionChange {
+    constructor(
+        /** Reference to the option that emitted the event. */
+        public source: OptionComponent,
+        /** Whether the change in the option's value was a result of a user action. */
+        public isUserInput = false) {
+    }
```

Comment:
What do you think about replacing `public` to `readonly`?

Reply:
make sense we are not going to change it

Reply:
Addressed